source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/Models/FeatureClassUtils.cs

```diff
@@ -361,18 +361,327 @@ private IFeatureClass ExportToShapefile(string fileNamePath, List<Graphic> graph
+
+                                field = new FieldClass();
+                                fieldEdit = (IFieldEdit)field;
+                                fieldEdit.Name_2 = "Distance";
+                                fieldEdit.AliasName_2 = "Geodetic Distance";
+                                fieldEdit.Type_2 = esriFieldType.esriFieldTypeDouble;
+                                fieldsEdit.AddField(field);
+
+                                field = new FieldClass();
+                                fieldEdit = (IFieldEdit)field;
+                                fieldEdit.Name_2 = "DistUnit";
+                                fieldEdit.AliasName_2 = "Distance Unit";
+                                fieldEdit.Type_2 = esriFieldType.esriFieldTypeString;
+                                fieldsEdit.AddField(field);
+
+                                field = new FieldClass();
+                                fieldEdit = (IFieldEdit)field;
+                                fieldEdit.Name_2 = "Radials";
+                                fieldEdit.AliasName_2 = "Radials";
+                                fieldEdit.Type_2 = esriFieldType.esriFieldTypeInteger;
+                                fieldsEdit.AddField(field);
+
+                                field = new FieldClass();
+                                fieldEdit = (IFieldEdit)field;
+                                fieldEdit.Name_2 = "CenterX";
+                                fieldEdit.AliasName_2 = "Center X";
+                                fieldEdit.Type_2 = esriFieldType.esriFieldTypeDouble;
+                                fieldsEdit.AddField(field);
+
+                                field = new FieldClass();
+                                fieldEdit = (IFieldEdit)field;
+                                fieldEdit.Name_2 = "CenterY";
+                                fieldEdit.AliasName_2 = "Center Y";
+                                fieldEdit.Type_2 = esriFieldType.esriFieldTypeDouble;
+                                fieldsEdit.AddField(field);
+
+                                break;
+                            }
+                    }
 
                     featClass = featureWorkspace.CreateFeatureClass(nameOfShapeFile, fields, null, null, esriFeatureType.esriFTSimple, shapeFieldName, "");
 
                     foreach (Graphic graphic in graphicsList)
                     {
+                        if (graphic.Attributes == null)
+                            continue;
                         IFeature feature = featClass.CreateFeature();
 
                         if (polyLineFC)
                             feature.Shape = graphic.Geometry;
                         else
                             feature.Shape = PolylineToPolygon(graphic.Geometry);
+                        switch (graphic.GraphicType.ToString())
+                        {
+                            case "Line":
+                                {
+                                    System.Object dist;
```

Comment:
Similar/same as code at: https://github.com/Esri/distance-direction-addin-dotnet/blob/dev/source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/Models/FeatureClassUtils.cs#L165 - suggest combining/reusing in case something would need changed